tests/Unit/Domain.LinnApps.Tests/ImportBookServiceTests/WhenUpdatingWithInvoiceDetails.cs

```diff
@@ -0,0 +1,60 @@
+﻿
+
+namespace Linn.Stores.Domain.LinnApps.Tests.ImportBookServiceTests
+{
+    using FluentAssertions;
+    using Linn.Stores.Domain.LinnApps.Parts;
+    using NSubstitute;
+    using NUnit.Framework;
+    using System;
+    using System.Collections.Generic;
+    using System.Linq;
+    using System.Linq.Expressions;
+
+    using Linn.Stores.Domain.LinnApps.ImportBooks;
+
+    public class WhenUpdatingWithInvoiceDetails : ContextBase
+    {
+        private static readonly int impbookId = 12007;
+        private ImportBook impbook;
+        private IEnumerable<ImportBookInvoiceDetail> invoiceDetails;
+
+        private ImportBookInvoiceDetail firstInvoiceDetail = new ImportBookInvoiceDetail() { ImportBookId = impbookId, InvoiceNumber = "123", LineNumber = 1 };
+        private ImportBookInvoiceDetail secondInvoiceDetail = new ImportBookInvoiceDetail() { ImportBookId = impbookId, InvoiceNumber = "1234", LineNumber = 2};
+
+
+        [SetUp]
+        public void SetUp()
+        {
+            this.impbook = new ImportBook()
+                               {
+                                   Id = impbookId
+            };
+            this.invoiceDetails = new List<ImportBookInvoiceDetail> { this.firstInvoiceDetail, this.secondInvoiceDetail };
+
+            var orderDetails = new List<ImportBookOrderDetail>();
+            var postEntries = new List<ImportBookPostEntry>();
+
+
+            //below doesn't actually work
```

Comment:
not finished yet obviously but all you need to do is set up this.impbook at the start of the test and check it again at the end of the test to see that all the changes you expect have happened (probably need a few tests to check updating, adding lines etc etc).